code/game/machinery/modular_shield.dm

```diff
@@ -783,3 +844,29 @@
 		return
 
 	shield_generator.shield_drain(15 / severity) //Light is 2 heavy is 1, note emp is usually a large aoe, tweak the number if not enough damage
+
+//Damage from explosions, remember we dont use armor
+/obj/structure/emergency_shield/modular/ex_act(severity)
+	if(isnull(shield_generator))
+		qdel(src)
+		return
```

Comment:
this should never happen, bug masking

Reply:
i did what a few people told me to and had the shields self qdel if they are initialized with no generator link and keeping this self qdel incase something happens like an admin presses buttons they shouldnt.